store/store.go

```diff
@@ -539,6 +561,33 @@ func (bs *BlockStore) Close() error {
 	return bs.db.Close()
 }
 
+// Rollback rollbacks the latest block from BlockStore.
+func (bs *BlockStore) Rollback() error {
```

Comment:
It would be great to have a test for this

Reply:
added the tests

```
$ go test ./store -run TestRollback -v
=== RUN   TestRollback
=== RUN   TestRollback/Successful_Rollback
=== RUN   TestRollback/Rollback_at_Height_Zero
=== RUN   TestRollback/Rollback_on_Empty_Store
=== RUN   TestRollback/Rollback_State_Consistency
--- PASS: TestRollback (0.01s)
    --- PASS: TestRollback/Successful_Rollback (0.00s)
    --- PASS: TestRollback/Rollback_at_Height_Zero (0.00s)
    --- PASS: TestRollback/Rollback_on_Empty_Store (0.00s)
    --- PASS: TestRollback/Rollback_State_Consistency (0.00s)
PASS
ok  	github.com/cometbft/cometbft/store	0.016s
```